NateGoSearch/NateGoSearchQuery.php

```diff
@@ -234,10 +234,14 @@ public function query($keywords)
 		$misspellings = $this->getPopularReplacements($keywords, $misspellings);
 
 		$keywords = $this->normalizeKeywordsForSearching($keywords);
-		$keywords_hash = sha1($keywords);
+		$unique_hash = sha1($keywords.serialize($this->document_types));
```

Comment:
@charleswaddell I've double-checked. This is an array in the form `[shortname => id, shortname2 => id2]`